client/src/pages/ThesisOverviewPage/ThesisOverviewPage.tsx

```diff
@@ -1,34 +1,57 @@
-import React from 'react'
-import { Container, Space } from '@mantine/core'
-import chartData from './mock/chart-data.json'
-import CustomThesisProgressChart from './components/CustomThesisProgressChart/CustomThesisProgressChart'
-import ThesisProgressFilter from './components/ThesisProgressFilter/ThesisProgressFilter'
-import ThesisOverviewChartProvider from './components/ThesisOverviewChartProvider/ThesisOverviewChartProvider'
-import { IThesisProgressChartDataElement } from './types/chart'
+import React, { useState } from 'react'
+import ThesesFilters from './components/ThesesFilters/ThesesFilters'
+import ThesesProvider from '../../contexts/ThesesProvider/ThesesProvider'
 import { usePageTitle } from '../../hooks/theme'
+import ContentContainer from '../../app/layout/ContentContainer/ContentContainer'
+import ThesesTable from '../../components/ThesesTable/ThesesTable'
+import ThesesGanttChart from '../../components/ThesesGanttChart/ThesesGanttChart'
+import { Button, Group, Space, Title } from '@mantine/core'
+import { ThesisState } from '../../requests/responses/thesis'
+import CreateThesisModal from './components/CreateThesisModal/CreateThesisModal'
+import { Plus } from 'phosphor-react'
+import { useHasGroupAccess } from '../../hooks/authentication'
 
 const ThesisOverviewPage = () => {
   usePageTitle('Theses Overview')
 
+  const [openCreateThesisModal, setOpenCreateThesisModal] = useState(false)
+
+  const managementAccess = useHasGroupAccess('admin', 'supervisor', 'advisor')
+
   return (
-    <Container size='xl'>
-      <ThesisOverviewChartProvider
-        mockData={{
-          thesisData: chartData.data as IThesisProgressChartDataElement[],
-          advisors: chartData.advisors,
-        }}
+    <ContentContainer>
+      <CreateThesisModal
+        opened={openCreateThesisModal}
+        onClose={() => setOpenCreateThesisModal(false)}
+      />
+      <ThesesProvider
+        fetchAll={true}
+        defaultStates={
+          managementAccess
+            ? [
+                ThesisState.PROPOSAL,
+                ThesisState.WRITING,
+                ThesisState.SUBMITTED,
+                ThesisState.ASSESSED,
+                ThesisState.GRADED,
+              ]
+            : [ThesisState.FINISHED]
```

Comment:
```suggestion
        defaultStates={}
```
Wouldn't it make sense to see all theses here by default?

General Question: 
How is this page relevant for students? While it’s true that other theses can be shared with students, introducing a new tab for this purpose might make sense due to its distinct use case.

Reply:
For advisors and supervisors this page will be used to check the progress and have an overview of all ongoing theses (this is a requirement from Stephan)

While implementing that, I thought why not make this page also available to everyone, especially since theses have a visibility setting (PRIVATE, INTERNAL, PUBLIC) which configures who can access it. Could be cool for students to browse finished theses that are marked as public.

Reply:
I see, I thought this page is meant to be something like a hub for supervisors and advisors. 

Maybe a more intuitive name for this tab would be a good enhancement. 
Something like "Browse Theses" or "All Theses"/"Active Theses"...still - I don't want to overrule any of @krusche's requirements.

Either way, from a usability perspective I would leave the filters empty by default.

Reply:
I have changed the page to be only visible for advisors and supervisors again as it was requested by Stephan as well.
There also was a discussion about the default states but we agreed to discuss this once the Gantt chart on that page is also implemented as we do not have the full picture right now